### PR TITLE
Add a note when trying to call a non-function

### PR DESCRIPTION
**Pull request by Detegr**
r? @Manishearth

### CODE REVIEW

**Comment by Detegr** [comment]
Let's see what the build bot says about this PR. One test failed on my machine even with the master branch of Rust.
So, the error + note now looks following:
```rust
test.rs:6:5: 6:10 error: expected function, found `&str`
test.rs:6     foo();
              ^~~~~
test.rs:5:9: 5:12 note: defined here
test.rs:5     let foo="foo";
```

However, when the non-function is a function argument, the span is a bit funny, as it points to the function argument, but I think it is still quite informative:
```rust
test.rs:2:5: 2:8 error: expected function, found `i32`
test.rs:2     i();
              ^~~
test.rs:1:9: 1:10 note: defined here
test.rs:1 fn func(i: i32) {
```

**Comment by Manishearth** [comment]
Looks good. Could you add a compile-fail test for this? (tests/compile-fail). Just some code that throws this error, with `//~ERROR foo` and `//~NOTE baz` to mark where the notes occur. Use `//~^ NOTE` in case you want to refer to a note that should occur on the previous line.

Run the test with `make check-stage1-cfail TESTNAME=name-of-test-file`

**Comment by Detegr** [comment]
Done. Should I rebase the PR to be a single commit only?

**Comment by Manishearth** [comment]
In this case two commits is fine.

But in general, feel free to rebase. `git commit --amend` is good too.

**Comment by Manishearth** [comment]
Do tests pass locally?

File: src/librustc_typeck/check/callee.rs [hunk 1]
```diff
@@ -231,6 +231,17 @@ fn confirm_builtin_call<'a,'tcx>(fcx: &FnCtxt<'a,'tcx>,
                 format!("expected function, found `{}`", actual)
             }, callee_ty, None);
 
+            if let hir::ExprCall(ref expr, _) = call_expr.node {
+                let tcx = fcx.tcx();
+                if let Some(pr) = tcx.def_map.borrow().get(&expr.id) {
+                    if pr.depth == 0 {
```
Comment (Manishearth) [inline src/librustc_typeck/check/callee.rs]: Good call on the depth check!

**Comment by Manishearth** [comment]
Thanks!

File: src/librustc_typeck/check/callee.rs (hunk 1 above)
Comment (Detegr) [inline src/librustc_typeck/check/callee.rs]: Thanks! Some of the tests caught that one.

**Pull request merged**